include/mbedtls/ssl_internal.h

```diff
@@ -523,74 +523,74 @@ static int mbedtls_ssl_parse_dns( const unsigned char *buf, size_t len )
         return( 1 );
     }
 
-    state = DNS_PARSER_STATE_START;
+    state = MBEDTLS_DNS_PARSER_STATE_START;
     i = 0;
     while( i < len )
     {
         switch( state )
         {
-        case DNS_PARSER_STATE_START:
+        case MBEDTLS_DNS_PARSER_STATE_START:
             if( buf[i] == ' ' )
             {
-                state = DNS_PARSER_STATE_ACCEPT_SPACE;
+                state = MBEDTLS_DNS_PARSER_STATE_ACCEPT_SPACE;
                 i++;
             }
             else if( isalpha( buf[i] ) )
             {
-                state = DNS_PARSER_STATE_ACCEPT_NOSPACE;
+                state = MBEDTLS_DNS_PARSER_STATE_ACCEPT_NOSPACE;
                 i++;
             }
             else
-                state = DNS_PARSER_STATE_REJECT;
+                state = MBEDTLS_DNS_PARSER_STATE_REJECT;
             break;
-        case DNS_PARSER_STATE_ACCEPT_SPACE:
-            state = DNS_PARSER_STATE_REJECT;
+        case MBEDTLS_DNS_PARSER_STATE_ACCEPT_SPACE:
+            state = MBEDTLS_DNS_PARSER_STATE_REJECT;
             break;
-        case DNS_PARSER_STATE_ACCEPT_NOSPACE:
+        case MBEDTLS_DNS_PARSER_STATE_ACCEPT_NOSPACE:
             if( isalpha( buf[i] ) || isdigit( buf[i] ) )
                 i++;
             else if( buf[i] == '-' )
             {
-                state = DNS_PARSER_STATE_DASH;
+                state = MBEDTLS_DNS_PARSER_STATE_DASH;
                 i++;
             }
             else if( buf[i] == '.' )
             {
-                state = DNS_PARSER_STATE_DOT;
+                state = MBEDTLS_DNS_PARSER_STATE_DOT;
                 i++;
             }
             else
-                state = DNS_PARSER_STATE_REJECT;
+                state = MBEDTLS_DNS_PARSER_STATE_REJECT;
             break;
-        case DNS_PARSER_STATE_DOT:
+        case MBEDTLS_DNS_PARSER_STATE_DOT:
             if( isalpha( buf[i] ) )
             {
-                state = DNS_PARSER_STATE_ACCEPT_NOSPACE;
+                state = MBEDTLS_DNS_PARSER_STATE_ACCEPT_NOSPACE;
                 i++;
             }
             else
-                state = DNS_PARSER_STATE_REJECT;
+                state = MBEDTLS_DNS_PARSER_STATE_REJECT;
             break;
-        case DNS_PARSER_STATE_DASH:
+        case MBEDTLS_DNS_PARSER_STATE_DASH:
```

Comment:
We don't seem to accept names with two consecutive hyphens and looking at the standard I believe that it should be possible.